pkg/sql/plan/function/baseTemplate.go

```diff
@@ -1609,6 +1609,54 @@ func opBinaryBytesBytesToFixedWithErrorCheck[Tr types.FixedSizeTExceptStrType](
 	return nil
 }
 
+func compareVarlenaEqual(parameters []*vector.Vector, result vector.FunctionResultWrapper, _ *process.Process, length int) error {
+	rs := vector.MustFunctionResult[bool](result)
+	rsVec := rs.GetResultVector()
+	rss := vector.MustFixedCol[bool](rsVec)
+
+	v1, v2 := parameters[0], parameters[1]
+	if v1.IsConstNull() || v2.IsConstNull() {
+		nulls.AddRange(rsVec.GetNulls(), 0, uint64(length))
+		return nil
+	}
+	col1, _ := vector.MustVarlenaRawData(v1)
+	col2, _ := vector.MustVarlenaRawData(v2)
+
+	if v1.IsConst() && v2.IsConst() {
+		p1 := col1[0].UnsafePtr()
```

Comment:
Do you really need unsafe pointe?      Even if you really have to this should be part of vector impl and should not escape the vector package.

Reply:
fixed